pkg/resource/provider/main_test.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2016-2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package provider
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+	"gotest.tools/v3/assert"
```

Comment:
We usually use `github.com/stretchr/testify/assert`

Reply:
Auto import had a mind of its own, good catch!